src/simulators/matrix_product_state/matrix_product_state_internal.cpp

```diff
@@ -349,6 +427,47 @@ void MPS::apply_diagonal_matrix(const AER::reg_t &qubits, const cvector_t &vmat)
   apply_matrix(qubits, diag_mat);
 }
 
+void MPS::centralize_qubits(MPS & new_MPS, const reg_t &qubits, 
+			    uint_t &new_first, uint_t &new_last, bool & ordered ) const {
+  new_MPS.initialize(*this);
+  ordered = true;
+
+  if (qubits.size() == 1) {
+    new_first = qubits[0];
+    new_last = qubits[0];
+    return;
+  }
+  std::vector<uint_t> internalIndexes;
+  for (uint_t index : qubits) {
+    internalIndexes.push_back(index);
+  }
+  
+  for (uint_t index=0; index < qubits.size()-1; index++) {
+    if (qubits[index] > qubits[index+1]){
+      ordered = false;
+      break;
+    }
+  }
+  if (!ordered)
+      sort(internalIndexes.begin(), internalIndexes.end());
+
+  std::vector<uint_t> new_indexes = calc_new_indexes(internalIndexes);
+  
+  uint_t avg = new_indexes[new_indexes.size()/2];
+  std::vector<uint_t>::iterator it = lower_bound(internalIndexes.begin(), internalIndexes.end(), avg);
```

Comment:
nit: There's a very useful type: `auto`, that will infere the type of the variable so you don't have to type: `std::vector<uint_t>::iterator`:
```c++
auto it = lower_bound(internalIndexes.begin(), internalIndexes.end(), avg);
```

Reply:
I agree. Changed it.